### DISCUSSION TITLE
FYI: Dependencies

### DISCUSSION THREAD

**Discussion by ryftjet** [J-J-B-J/get-sentral#41] [General]
Hi there!
We have been criticised by some for our extremely long list of dependencies, but they are all required for the operation of get-sentral. Here's how:

| Dependency | Use |
|--------------|-----|
| beautifulsoup4 | Scraping the Sentral data |
| pillow | Loading the image files for the bouncing Sentral logo |
| pyinstaller | Saving the app files to your computer |
| paperclip | Copying URLs from the GUI |
| python-dotenv | Loading credentials from a .env file |
| selenium | Logging into Sentral and navigating around |
| setuptools | Setting up get-sentral as a package |
| BarcodeGenerator | Generating the library barcode |
| webdriver_manager | Used for Selenium |

**Comment by SuperHarmony910** [comment]
who criticised us for having a few dependencies lol
